### PR TITLE
docker builder script; package_all remade to acomodate new builder

### PR DESCRIPTION
**Pull request by zolia**

### CODE REVIEW

File: bin/util_docker/builder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM golang:1.9.2 AS builder
+
+ARG PACKAGE_VERSION
```
Comment (Waldz) [inline bin/util_docker/builder/Dockerfile]: Not needed

File: bin/package_all [hunk 2]
```diff
@@ -22,33 +22,24 @@ printf "Creating build directory '$DIR_BUILD' ..\n" \
...
     && mv build/client/mysterium_client $DIR_BUILD/mysterium_client_linux_amd64 \
 && printf "\n\n" \
-    && GOOS=linux GOARCH=386 bin/client_build \
-    && bin/client_package_debian $VERSION i386 \
+    && bin/package_builder "bin/client_build; bin/client_package_debian" $VERSION i386 \
     && mv build/client/mysterium_client $DIR_BUILD/mysterium_client_linux_386 \
 && printf "\n\n" \
-    && GOOS=linux GOARCH=arm bin/client_build \
-    && bin/client_package_debian $VERSION armhf \
+    && bin/package_builder "bin/client_build; bin/client_package_debian" $VERSION armhf \
```
Comment (tadovas) [inline bin/package_all]: What about moving these a lot of building lines for different architectures into single loop? for example:
for each arch in architectures: package server, package client

File: bin/package_builder_push [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    DOCKER_TAG=$VERSION
+fi
+
+printf "Building Docker image '$DOCKER_IMAGE:$DOCKER_TAG' ..\n" \
+    && docker build \
+        --file bin/util_docker/builder/Dockerfile \
+        --build-arg PACKAGE_VERSION=$VERSION \
+        --tag $DOCKER_IMAGE:$DOCKER_TAG \
+        . \
+    && docker push $DOCKER_IMAGE:$DOCKER_TAG
```
Comment (Waldz) [inline bin/package_builder_push]: Could be align this script in `server_package_docker` 
- `builder_package` - builds just local image
- `builder_package latest` - builds && pushes to Docker hub

File: bin/package_builder [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# Usage example:
+#> bin/server_package_deb 0.0.1 amd64
+
+COMMAND=$1
+if [ -z "$COMMAND" ]; then
+    printf "\e[0;31m%s\e[0m\n" "Missing command!"
+    exit 1
+fi
+
+VERSION=$2
```
[the remark this answers is not in the document]
Comment (Waldz) [reply bin/package_builder]: **CONTRIBUTING.md -> Debian packaging** could be updated now

File: bin/util_docker/builder/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
+FROM golang:1.9.2 AS builder
```
Comment (Waldz) [inline bin/util_docker/builder/Dockerfile]: Would be nice to use this new image in `bin/server_docker/ubuntu/Dockerfile` etc.

Comment (zolia) [reply bin/util_docker/builder/Dockerfile]: changed; tested

File: bin/builder_package [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+# Usage - build image and push to docker hub:
+#> bin/builder_package latest
+
+DOCKER_IMAGE="mysteriumnetwork/mysterium-node-builder"
+
+printf "Building Docker image '$DOCKER_IMAGE' ..\n" \
+    && docker build \
+        --file bin/util_docker/builder/Dockerfile \
+        --build-arg PACKAGE_VERSION=latest \
```
Comment (Waldz) [inline bin/builder_package]: Not needed argument

File: bin/client_docker/ubuntu/Dockerfile [hunk 7]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.9.2 AS builder
+FROM mysteriumnetwork/mysterium-node-builder AS builder
```
Comment (Waldz) [inline bin/client_docker/ubuntu/Dockerfile]: I implemented this in https://github.com/MysteriumNetwork/node/pull/202/files

**Pull request merged**